File: pulse/pulse/report/pulse_workload/pulse_workload.js
frappe.query_reports['Pulse Workload'] = {
  filters: [
    {fieldname: 'project', label: __('Project'), fieldtype: 'Link', options: 'Project'},
    {fieldname: 'start_date', label: __('From'), fieldtype: 'Date', default: frappe.datetime.get_today(), reqd: 1},
    {fieldname: 'end_date', label: __('To'), fieldtype: 'Date', default: frappe.datetime.add_days(frappe.datetime.get_today(), 13), reqd: 1},
  ],
};

File: pulse/pulse/report/pulse_workload/pulse_workload.py
"""Permission-scoped estimated work compared with allocation and leave capacity."""
from frappe.utils import today, add_days
from pulse.api.business import workload


def execute(filters=None):
    filters = filters or {}
    start = filters.get('start_date') or today()
    end = filters.get('end_date') or add_days(start, 13)
    result = workload(start, end, filters.get('project'))
    columns = [
        {'label': 'Team member', 'fieldname': 'user', 'fieldtype': 'Data', 'width': 220},
        {'label': 'Estimated hours', 'fieldname': 'estimated_hours', 'fieldtype': 'Float', 'width': 150},
        {'label': 'Available hours', 'fieldname': 'available_hours', 'fieldtype': 'Float', 'width': 150},
        {'label': 'Over capacity (hours)', 'fieldname': 'over_hours', 'fieldtype': 'Float', 'width': 170},
    ]
    return columns, result['rows']
